src/main/java/gregtech/api/recipes/ModHandler.java

```diff
@@ -703,26 +779,32 @@ public static void removeRecipeByName(String recipeName) {
      * @param startTier  The starting tier index, inclusive.
      * @param endTier    The ending tier index, inclusive.
      */
-    public static void removeTieredRecipeByName(String recipeName, int startTier, int endTier) {
-        for (int i = startTier; i <= endTier; i++)
+    @SuppressWarnings("unused")
+    public static void removeTieredRecipeByName(@Nonnull String recipeName, int startTier, int endTier) {
+        for (int i = startTier; i <= endTier; i++) {
             removeRecipeByName(String.format("%s%s", recipeName, GTValues.VN[i].toLowerCase()));
+        }
     }
 
     ///////////////////////////////////////////////////
     //            Get Recipe Output Helpers          //
     ///////////////////////////////////////////////////
 
-    public static Pair<IRecipe, ItemStack> getRecipeOutput(World world, ItemStack... recipe) {
-        if (recipe == null || recipe.length == 0)
-            return ImmutablePair.of(null, ItemStack.EMPTY);
-
+    /**
+     * @param world  the world to check the output from
+     * @param recipe the recipe to retrieve from
+     * @return a Pair of the recipe, and the output
+     */
+    @Nonnull
+    public static Pair<IRecipe, ItemStack> getRecipeOutput(@Nullable World world, @Nonnull ItemStack... recipe) {
+        if (recipe.length == 0) return ImmutablePair.of(null, ItemStack.EMPTY);
```

Comment:
This could technically still be null, since it is an array of item stacks. I would keep the null check here just in case.